crates/turborepo-wax/src/walk.rs

```diff
@@ -570,9 +578,22 @@ pub struct Walk<'g> {
     root: PathBuf,
     prefix: PathBuf,
     walk: walkdir::IntoIter,
+    // This is a hack to express an empty iterator
+    is_empty: bool,
```

Comment:
This isn't blocking, but I feel like we could either make this into an `Option` or have `walk::walk` return a generic `impl Iterator` and then return an empty iterator instead.

Reply:
Agreed, this is definitely not the optimal way of doing this.